app/src/org/commcare/activities/DrawActivity.java

```diff
@@ -205,6 +207,8 @@ private void saveAndClose() {
             setResult(AppCompatActivity.RESULT_OK);
         } catch (FileNotFoundException e) {
             e.printStackTrace();
+            // We shouldn't have gotten a file not found exception since we're using ImageWidget.getTempFileForImageCapture.
+            Logger.log(LogTypes.TYPE_ERROR_DESIGN, "Couldn't save signature at " + output.toString() + " because of" + e.getMessage());
```

Comment:
can `output` be null ?

Reply:
That's a good question. Umm, Although it gets set in `onCreate` [here](https://github.com/dimagi/commcare-android/blob/30f899656c672b22e7784a49eac8287ce2d2b277/app/src/org/commcare/activities/DrawActivity.java#L96) and [here](https://github.com/dimagi/commcare-android/blob/30f899656c672b22e7784a49eac8287ce2d2b277/app/src/org/commcare/activities/DrawActivity.java#L134-L136) but it could get null maybe. 

Though, I think the `FileNotFoundException` exception would only be thrown when the `output` is non-null. Otherwise it'd have thrown an NPE right? So I'm assuming that it's safe to access it here.

Reply:
sounds good